examples/complete-http/main.tf

```diff
@@ -1,323 +1,262 @@
-  }
-}
-
-########################
-# AWS Cognito User Pool
-########################
+################################################################################
+# Supporting Resources
+################################################################################
 
 resource "aws_cognito_user_pool" "this" {
-  name = "user-pool-${random_pet.this.id}"
-}
+  name = local.name
 
-####################
-# AWS Step Function
-####################
+  tags = local.tags
+}
 
 module "step_function" {
   source  = "terraform-aws-modules/step-functions/aws"
-  version = "~> 2.0"
-
-  name = random_pet.this.id
-
-  definition = <<EOF
-{
-  "Comment": "A Hello World example of the Amazon States Language using Pass states",
-  "StartAt": "Hello",
-  "States": {
-    "Hello": {
-      "Type": "Pass",
-      "Result": "Hello",
-      "Next": "World"
-    },
-    "World": {
-      "Type": "Pass",
-      "Result": "World",
-      "End": true
-    }
-  }
-}
-EOF
-}
-
-##################
-# Extra resources
-##################
-
-resource "random_pet" "this" {
-  length = 2
-}
-
-#############################################
-# Using packaged function from Lambda module
-#############################################
+  version = "~> 4.0"
 
-locals {
-  package_url = "https://raw.githubusercontent.com/terraform-aws-modules/terraform-aws-lambda/master/examples/fixtures/python3.8-zip/existing_package.zip"
-  downloaded  = "downloaded_package_${md5(local.package_url)}.zip"
-}
+  name      = local.name
+  role_name = "${local.name}-step-function"
+  trusted_entities = [
+    "apigateway.amazonaws.com",
+  ]
 
-resource "null_resource" "download_package" {
-  triggers = {
-    downloaded = local.downloaded
+  attach_policies_for_integrations = true
+  service_integrations = {
+    stepfunction = {
+      stepfunction = ["*"]
+    }
   }
 
-  provisioner "local-exec" {
-    command = "curl -L -o ${local.downloaded} ${local.package_url}"
-  }
+  definition = <<-EOT
+    {
+      "Comment": "A Hello World example of the Amazon States Language using Pass states",
+      "StartAt": "Hello",
+      "States": {
+        "Hello": {
+          "Type": "Pass",
+          "Result": "Hello",
+          "Next": "World"
+        },
+        "World": {
+          "Type": "Pass",
+          "Result": "World",
+          "End": true
+        }
+      }
+    }
+  EOT
+
+  tags = local.tags
 }
 
 module "lambda_function" {
   source  = "terraform-aws-modules/lambda/aws"
-  version = "~> 3.0"
+  version = "~> 7.0"
 
-  function_name = "${random_pet.this.id}-lambda"
+  function_name = local.name
   description   = "My awesome lambda function"
-  handler       = "index.lambda_handler"
-  runtime       = "python3.8"
+  handler       = "lambda.handler"
+  runtime       = "python3.12"
+  architectures = ["arm64"]
+  publish       = true
+  source_path   = "lambda.py"
 
-  publish = true
-
-  create_package         = false
-  local_existing_package = local.downloaded
```

Comment:
Oh, I didn't know a single python file would require the Python runtime. Yes, I can absolutely revert - but let me update the lambda module to use a newer version since 3.8 is slated to reach EOL this year https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

Reply:
xref https://github.com/terraform-aws-modules/terraform-aws-lambda/pull/573

Looks like the zip archive is not dependent on Python3.8, but the name sort of implies it requires Python3.8 which is reaching EOL this year

Reply:
fixed in https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/pull/54/commits/d6e694c0e73bf8c7fc7d145e074c3d562dc36848